core: frontend: do not compare non-semver tags to semver tags

Williangalvani opened the pull request:
fix #1423

joaoantoniocardoso commented on core/frontend/src/components/version-chooser/VersionChooser.vue:
```diff
@@ -330,6 +330,12 @@ export default Vue.extend({
+      } catch {
+        // if we are on a random, non-SemVer tag, we cant compare it to a SemVer tag.
+        return ''
+      }
```
The compiler seems skeptical about the need for this try-catch

Williangalvani replied:
got rid of it

joaoantoniocardoso reviewed (approved):
Tested, and I can confirm that it solves the problem :)

just make the compiler happy and you're good to go

joaoantoniocardoso reviewed (changes requested):
Actually, there is a missing detail here:
- When we are at the latest master, the spinner keeps spinning forever
![image](https://user-images.githubusercontent.com/5920286/216202310-03e264c9-1d8b-476d-8c14-a78417b9de30.png)

Williangalvani commented:
updated and rebased, no spinner for me:
<img width="950" alt="Screenshot 2023-02-02 at 12 27 56" src="https://user-images.githubusercontent.com/4013804/216367451-fee9a69e-6c75-4fb1-b5a6-3b9efe00b72f.png">

joaoantoniocardoso reviewed (approved):
Confirmed :)
